doc/source/whatsnew/v0.23.0.txt

```diff
@@ -119,6 +119,51 @@ Current Behavior
 
     s.rank(na_option='top')
 
+.. _whatsnew_0230.enhancements.round-trippable_json:
+
+JSON read/write round-trippable with ``orient='table'``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A ``DataFrame`` can now be written to and subsequently read back via JSON while preserving metadata through usage of the ``orient='table'`` argument (see :issue:`18912` and :issue:`9146`). Previously, none of the available ``orient`` values guaranteed the preservation of dtypes and index names, amongst other metadata.
+
+.. ipython:: python
+
+   df = pd.DataFrame({'foo': [1, 2, 3, 4],
+		      'bar': ['a', 'b', 'c', 'd'],
+		      'baz': pd.date_range('2018-01-01', freq='d', periods=4),
+		      'qux': pd.Categorical(['a', 'b', 'c', 'c'])
+		      }, index=pd.Index(range(4), name='idx'))
+   df
+
+Previous Behavior:
+
+.. code-block:: ipython
+
+   In [17]: df.to_json("test.json", orient='columns')
+   In [17]: pd.read_json("test.json", orient='columns')
+   Out[18]:
+       bar          baz  foo qux
+   0   a  1514764800000    1   a
+   1   b  1514851200000    2   b
+   2   c  1514937600000    3   c
+   3   d  1515024000000    4   c
+
+Current Behavior:
+
+.. code-block:: ipython
+
+   In [29]: df.to_json("test.json", orient='table')
+   In [30]: pd.read_json("test.json", orient='table')
+   Out[30]:
+       bar        baz  foo qux
+   idx
+   0     a 2018-01-01    1   a
+   1     b 2018-01-02    2   b
+   2     c 2018-01-03    3   c
+   3     d 2018-01-04    4   c
+
+Please note that the string `index` is not supported with the round trip format, as it is used by default in ``write_json`` to indicate a missing index name.
```

Comment:
this should be in the doc-string / io.rst doc section as well (and add this example there too)